ArduSub/control_althold.cpp

```diff
@@ -103,33 +103,18 @@ void Sub::althold_run()
 }
 
 void Sub::control_depth() {
-    // Hold actual position until zero derivative is detected
-    static bool engageStopZ = true;
-    // Get last user velocity direction to check for zero derivative points
-    static bool lastVelocityZWasNegative = false;
-    if (fabsf(channel_throttle->norm_input()-0.5f) > 0.05f) { // Throttle input above 5%
-        // output pilot's throttle
-        attitude_control.set_throttle_out(channel_throttle->norm_input(), false, g.throttle_filt);
-        // reset z targets to current values
-        pos_control.relax_alt_hold_controllers();
-        engageStopZ = true;
-        lastVelocityZWasNegative = is_negative(inertial_nav.get_velocity_z());
-    } else { // hold z
 
-        if (ap.at_bottom) {
-            pos_control.relax_alt_hold_controllers(); // clear velocity and position targets
-            pos_control.set_alt_target(inertial_nav.get_altitude() + 10.0f); // set target to 10 cm above bottom
-        }
+    pos_control.update_z_controller();
 
-        // Detects a zero derivative
-        // When detected, move the altitude set point to the actual position
-        // This will avoid any problem related to joystick delays
-        // or smaller input signals
-        if(engageStopZ && (lastVelocityZWasNegative ^ is_negative(inertial_nav.get_velocity_z()))) {
-            engageStopZ = false;
-            pos_control.relax_alt_hold_controllers();
-        }
+    motors.set_throttle(pos_control.get_throttle() + channel_throttle->norm_input());
```

Comment:
Looks like you're removing the passthrough of the throttle through the attitude controller.  Just because you don't use it in this particular sub frame doesn't mean it's the right thing to do! :-).

Reply:
This was actually a mistake, thank you for the review!